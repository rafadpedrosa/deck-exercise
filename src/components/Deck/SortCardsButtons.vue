<template>
  <div class="sort-cards-buttons">
    <div class="title">Sort cards by</div>
    <div>
      <button
        data-test="sortByInput"
        class="primary-color"
        @click="$emit('sortByInput')"
      >
        Input Order
      </button>

      <button
        data-test="sortByRotation"
        class="primary-color"
        @click="$emit('sortByRotation')"
      >
        Rotation order
      </button>
    </div>
  </div>
</template>

<script>
export default {
  data() {
    return {};
  },
};
</script>

<style scoped>
.title {
  padding: 5px;
}

.sort-cards-buttons {
  display: flex;
  flex-direction: column;
  justify-content: center;
}

input[type="text"] {
  border-radius: 4px;
}

button {
  width: 130px;
  height: 35px;
  border-radius: 2px;
  border: 1px solid #ccc;
}
</style>
