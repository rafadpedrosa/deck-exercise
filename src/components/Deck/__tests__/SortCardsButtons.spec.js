import { shallowMount } from "@vue/test-utils";
import SortCardsButtons from "../SortCardsButtons";

describe("SortCardsButtons.js", () => {
  it("Should create a snapshot of the SortCardsButtons", () => {
    const wrapper = shallowMount(SortCardsButtons, {});

    expect(wrapper.element).toMatchSnapshot();
  });

  it("Should emit sortByInput and sortByRotation after button click", () => {
    const wrapper = shallowMount(SortCardsButtons, {});
    wrapper.find("button[data-test='sortByRotation']").trigger("click");
    wrapper.find("button[data-test='sortByInput']").trigger("click");

    expect(wrapper.emitted("sortByInput")).toHaveLength(1);
    expect(wrapper.emitted("sortByRotation")).toHaveLength(1);
  });
});
