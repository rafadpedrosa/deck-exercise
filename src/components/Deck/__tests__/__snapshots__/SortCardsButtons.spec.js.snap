// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortCardsButtons.js Should create a snapshot of the SortCardsButtons 1`] = `
<div
  class="sort-cards-buttons"
>
  <div
    class="title"
  >
    Sort cards by
  </div>
   
  <div>
    <button
      class="primary-color"
      data-test="sortByInput"
    >
      
      Input Order
    
    </button>
     
    <button
      class="primary-color"
      data-test="sortByRotation"
    >
      
      Rotation order
    
    </button>
  </div>
</div>
`;
